src/actionTypes.js

```diff
@@ -405,10 +406,39 @@ type EventUserRemoveAction = {|
   // the type before going and adding those other properties here properly.
 |};
 
+/** Sent when editing a custom profile field */
+type UserUpdatePayloadCustomProfileField = {|
+  custom_profile_field: {
+    id: number,
+    rendered_value: string,
+    value: string,
+  },
+|};
+
+/** Sent when editing full name */
+type UserUpdatePayloadName = {|
+  email: string,
+  full_name: string,
+|};
+
+/** Sent when changing user's avatar */
+type UserUpdatePayloadAvatar = {|
+  avatar_source: UserAvatarSource,
+  avatar_url: string,
+  avatar_url_medium: string,
+  email: string,
+|};
+
+/** Contains only one of the possible payloads */
```

Comment:
Hey @borisyankov, I'm editing #3421 based on this PR. It seems that Zulip server API allows an update request to have multiple payloads, email and full name for example, so is there any reason to keep `UserUpdatePayload` to have only one payload?